### ISSUE TITLE
Unable to validate installation id

### ISSUE THREAD

**Issue by fbouchet** [bitwarden/core#172]
Hello,

I'm trying to install on a Synology NAS using an SSH connection and I'm bumping into an issue very early on in the installation process, when typing `./bitwarden install `

Docker and docker-compose are installed and detected just fine (`Docker version 17.05.0-ce, build 9f07f0e-synology` and `docker-compose version 1.14.0, build c7bdf9e`). After having provided my domain name and said "no" to the question about using "Let's encrypt", I'm asked to provide my installation id and key (which I retrieved, as requested, from bitwarden.com/host). After having given that information, the script tries to contact bitwarden servers and eventually times out with a message saying `Unable to validate installation id. Problem contacting bitwarden server`

I have tried pinging the address it's trying to contact ([https://api.bitwarden.com/installations](https://api.bitwarden.com/installations)) and I get a reply just fine, so I'm not sure why it would fail. From what I understand of [that file](https://github.com/bitwarden/core/blob/master/util/Setup/Program.cs), it is also not an issue relative to a typo in my installation id or key (the error message would be different).

Any suggestion?

**Comment by kspearrin** [comment]
Is there some firewall blocking access to the outside world from within the docker container? Have you tried the ping from actually inside the docker container itself rather than just on the host machine?

**Comment by fbouchet** [comment]
Thanks, I believe you're correct in your assessment. The script is launching docker with the `bridge` interface instead of `host`, and all my attempts to ping from a container using `bridge` have failed to connect to any outside server. Clearly this is on me, but so far my attempts to edit the default `bridge` information have failed, because the Synology docker configuration seems to differ from the standard one, so all the solutions I find online do not seem to apply.

I have noticed however that when I launch a docker container with the `--net=host` option, it seems to solve my problem and I can ping just fine. So if there were a way to add this to the script, that would probably solve my issue. I have tried to edit manually the `install.sh` file that is downloaded into `bwdata\scripts` to add this flag, but launching directly that script (even as it is downloaded) results in an error saying `docker: Error response from daemon: create ../.: "../." includes invalid characters for a local volume name (...). If you intended to pass a host directory, use absolute path.`

**TL;DR:** My question at this point is: would there be a way to make `install.sh` runnable after it's downloaded? As I think I could find a way to fix my original issue if that were possible.
The other option would be to add directly that flag to the downloaded file, but forcing to use `host` with docker might be an issue in some cases so I'd understand if you wouldn't want to do that.

**Comment by fbouchet** [comment]
Never mind: I just had to edit the `OUTPUT_DIR` at the beginning of the file to replace the value (`../.`) by the absolute path for my installation and the rest of the process worked fine. It seems it's the only element that prevented from running the script directly.
Also, I can confirm that adding `--net=host` when starting `docker` fixed the issue in my case.

Thanks again for pointing me in the right direction!

**Comment by fbouchet** [comment]
Unfortunately, I'm bumping into another issue (which I assume is related) later on in the process, when running `./bitwarden.sh start`
In order to be consistent with my aforementioned fix, I have edited `./bwdata/scripts/run.sh` to add `--net=host` after every call to `docker` well as the docker-compose file (`./bwdata/docker/docker-compose.yml`) to add to each of the services the corresponding line `network_mode: "host"`
(note: before doing that, I tried with the original file and I was having additional errors that got fixed)

So at this point, here is the output I obtain:
`
Pulling web (bitwarden/web:latest)...
latest: Pulling from bitwarden/web
Digest: sha256:526cd2b0d70f97930d3eb3d023241b23a608e03410c3a4c0861a44beb7e668da
Status: Image is up to date for bitwarden/web:latest
Pulling attachments (bitwarden/attachments:latest)...
latest: Pulling from bitwarden/attachments
Digest: sha256:d658685d00af29914dea150e755b5ba87e5fc8a891c493caf0ee88565b3800a8
Status: Image is up to date for bitwarden/attachments:latest
Pulling icons (bitwarden/icons:latest)...
latest: Pulling from bitwarden/icons
Digest: sha256:ba5008134c085a60d2d2766a844d531e49585835143aed1b98e228d2383bc57f
Status: Image is up to date for bitwarden/icons:latest
Pulling nginx (bitwarden/nginx:latest)...
latest: Pulling from bitwarden/nginx
Digest: sha256:c5278379ad94870be713f8e553bbdb23c252398556d48385df0053dfb03b3bc3
Status: Image is up to date for bitwarden/nginx:latest
Pulling api (bitwarden/api:latest)...
latest: Pulling from bitwarden/api
Digest: sha256:1f086e87a3324af886f7d1b9e70cea4170fb76885676554941c0e0dfdcbc9c7f
Status: Image is up to date for bitwarden/api:latest
Pulling mssql (bitwarden/mssql:latest)...
latest: Pulling from bitwarden/mssql
Digest: sha256:59c619900a68515ae82973ec8c73608c94d8d06770e3473e01831ce972fda5d5
Status: Image is up to date for bitwarden/mssql:latest
Pulling identity (bitwarden/identity:latest)...
latest: Pulling from bitwarden/identity
Digest: sha256:e634a3f3c9481ff81dcedea16a5ddd457372ebd40fcf9a74a340c0d7ffda21be
Status: Image is up to date for bitwarden/identity:latest
Creating api ...
Creating icons ...
Creating nginx ...
Creating web ...
Creating attachments ...
Creating mssql ...
Creating icons
Creating nginx
Creating identity ...
Creating web
Creating api
Creating attachments
Creating mssql
Creating api ... error
Creating nginx ... error
Creating attachments ... error
Creating mssql ... error
Creating identity ... error

ERROR: for identity  Cannot start service identity: Bind mount failed: '/var/services/homes/fb/bwdata/core' does not exists
ERROR: for nginx  Cannot start service nginx: Bind mount failed: '/var/services/homes/fb/bwdata/letsencrypt' does not exists
ERROR: for api  Cannot start service api: Bind mount failed: '/var/services/homes/fb/bwdata/core' does not exists
ERROR: for attachments  Cannot start service attachments: Bind mount failed: '/var/services/homes/fb/bwdata/core/attachments' does not exists
ERROR: for identity  Cannot start service identity: Bind mount failed: '/var/services/homes/fb/bwdata/core' does not exists
ERROR: for mssql  Cannot start service mssql: Bind mount failed: '/var/services/homes/fb/bwdata/mssql/backups' does not exists
ERROR: Encountered errors while bringing up the project.
`

Any thoughts on what could be the issue here?

**Comment by Snowrisk** [comment]
Just create the directories:
bwdata/core
bwdata/core/attachments
bwdata/letsencrypt
bwdata/mssql

**Comment by fbouchet** [comment]
It was indeed that simple - thank you very much Snowrisk.

Unfortunately, I'm getting an issue as well in the last step of the process when running `./bitwarden.sh updatedb`. Here is what I obtain:

Unhandled Exception: System.Data.SqlClient.SqlException: A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 35 - An internal exception was caught) ---> System.AggregateException: One or more errors occurred. (No such device or address) ---> System.Net.Internals.SocketExceptionFactory+ExtendedSocketException: No such device or address

It sounds similar to [#148](https://github.com/bitwarden/core/issues/148), but when I check the log in `./bwdata/core/logs/Api`, what I see is slightly different:

2017-12-28 17:54:21.559 +00:00 [Fatal] Unable to start Kestrel.
System.IO.IOException: Failed to bind to address http://[::]:80: address already in use. ---> Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.Internal.AddressInUseException: Error -98 EADDRINUSE address already in use ---> Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.UvException: Error -98 EADDRINUSE address already in use at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.LibuvFunctions.ThrowError(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.UvTcpHandle.GetSockIPEndPoint()
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Listener.ListenTcp(Boolean useFileHandle)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Listener.<>c.<StartAsync>b__8_0(Listener listener)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.LibuvThread.DoPostWork()

Besides, the answer provided by kspearrin back then points towards a section that no longer seems to exist, so it might be that the same solution (increasing the resources for docker) could work for me, but I'm not sure how to do that.

Any idea?

**Comment by kspearrin** [comment]
Closing this issue since it is stale and the original problem was resolved. Please open a new issue if you still have problems.
